scipy/sparse/sparsetools/py3k.h

```diff
@@ -1,23 +0,0 @@
-/*
- * Undefine macros defined by SWIG
- */
-#if PY_VERSION_HEX >= 0x03000000
-
-#ifdef PyInt_Check
-#undef PyInt_Check
-#endif
-
-static int __pyfile_check_guard(PyObject *x)
-{
-    return 0;
-}
-#define PyFile_Check(x) __pyfile_check_guard((x))
-static int __pyinstance_check_guard(PyObject *x)
-{
-    return 0;
-}
-#define PyInstance_Check(x) __pyinstance_check_guard((x))
-
-#endif
-
-#include "numpy/npy_3kcompat.h"
```

Comment:
This is no longer included which concerns me because the code compiles and runs on my computer.

I don't see a file by this name in this repository but I assume it's somehow referencing numpy via a submodule? or did a file by this name exist and was deleted at somepoint?

Reply:
This should be using `numpy/core/include/numpy/npy_3kcompat.h`, but it isn't clear to me why that file is found without `numpy.get_include()` being used to add to `include_dirs`

Reply:
Checked the build log, and it has the right include dir:
```
creating build/temp.linux-x86_64-3.7/scipy/sparse/sparsetools
compile options: '-D__STDC_FORMAT_MACROS=1 -Iscipy/sparse/sparsetools -I/home/rgommers/anaconda3/lib/python3.7/site-packages/numpy/core/include -I/home/rgommers/anaconda3/include/python3.7m -c'
```
Unclear why though:
```
    config.add_extension('_sparsetools',
                         define_macros=[('__STDC_FORMAT_MACROS', 1)],
                         depends=depends,
                         include_dirs=['sparsetools'],
```

It looks like the numpy include dir is always included, and in the extensions that have `include_dirs=[get_numpy_include_dirs()],` or similar it's included _twice_. I think that wasn't always the case, but on the other hand I don't see how this would have worked before. Not sure what's going on with that.

Reply:
A number of other places[1] seem to include this file. Would you like me to leave the file with just the include line to be safe or continue with the deletion assuming nothing is wrong (because the tests pass).


[1] scipy/ndimage/src/_ctest.c:
scipy/ndimage/src/nd_image.{c,h}
scipy/odr/odrpack.h:
scipy/signal/lfilter.c.src:
scipy/integrate/_odepackmodule.c:
scipy/sparse/linalg/dsolve/_superlu{module,_utils,object}.c:

Reply:
I expect numpy will want to remove that file, and it's kinda not public. So yes, I think it would be better to get rid of it everywhere.

It would be useful to at least do some sanity checks/greps that the defines in `npy_3kcompat.h` aren't used anywhere.

Reply:
The file ([npy_3kcompat.h](https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/npy_3kcompat.h
)) is pretty large. A quick scan suggests we aren't using any in this directory and I can work on cleaning up uses in other directories (scipy.ndimage, scipy.spare, and scipy.signal mostly) after this goes in.